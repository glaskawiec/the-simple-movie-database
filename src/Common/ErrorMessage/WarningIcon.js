import styled from 'styled-components';

const WarningIcon = styled.div`
  font-size: 3em;
`;

export default WarningIcon;
