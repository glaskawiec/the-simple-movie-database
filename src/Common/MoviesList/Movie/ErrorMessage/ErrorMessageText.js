import styled from 'styled-components';

const ErrorMessageText = styled.div`
  font-size: 1.5em;
`;

export default ErrorMessageText;
