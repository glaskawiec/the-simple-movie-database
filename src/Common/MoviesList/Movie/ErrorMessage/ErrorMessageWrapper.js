import styled from 'styled-components';

const ErrorMessageWrapper = styled.div`
  text-align: center;
  color: ${props => props.theme.primary};
  padding: 2em;
`;

export default ErrorMessageWrapper;
