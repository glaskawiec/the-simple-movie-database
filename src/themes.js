const Themes = {
  light: {
    colors: {
      primary: '#F2355B',
      secondary: '#161616',
      background: '#FFFFFF',
      text: '#000000',
      mobileMenuOverlay: '#000000',
      card: '#FFFFFF',
      themeToggleThumb: '#161616',
      paginationHover: '#CDCDCD',
    },
  },
  dark: {
    colors: {
      primary: '#F2355B',
      secondary: '#090909',
      background: '#161616',
      text: 'hsla(0,0%,100%,0.88)',
      mobileMenuOverlay: '#FFFFFF',
      card: '#090909',
      themeToggleThumb: '#FFFFFF',
      paginationHover: '#292929',
    },
  },
};

export default Themes;
