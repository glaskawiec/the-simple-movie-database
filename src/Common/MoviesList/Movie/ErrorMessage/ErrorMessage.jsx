import React from 'react';
import ErrorMessageText from './ErrorMessageText';
import WarningIcon from './WarningIcon';
import ErrorMessageWrapper from './ErrorMessageWrapper';

const ErrorMessage = () => (
  <ErrorMessageWrapper>
    <WarningIcon>âš </WarningIcon>
    <ErrorMessageText>Sorry, something went wrong.</ErrorMessageText>
    <ErrorMessageText>Please try again leter.</ErrorMessageText>
  </ErrorMessageWrapper>
);

ErrorMessage.propTypes = {
};

export default ErrorMessage;
