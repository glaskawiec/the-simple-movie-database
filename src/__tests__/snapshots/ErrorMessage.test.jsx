import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import ErrorMessage from '../../common/errorMessage/ErrorMessage';
import themes from '../../themes';

test('ErrorMessage snapshot', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={themes.light}>
        <ErrorMessage />
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
