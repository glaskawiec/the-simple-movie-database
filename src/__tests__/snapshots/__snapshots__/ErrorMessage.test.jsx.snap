// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage snapshot 1`] = `
<div
  className="sc-htpNat hFCDdZ"
>
  <div
    className="sc-bwzfXH ilocbC"
  >
    âš 
  </div>
  <div
    className="sc-bdVaJa dAIgSr"
  >
    Sorry, something went wrong.
  </div>
  <div
    className="sc-bdVaJa dAIgSr"
  >
    Please try again leter.
  </div>
</div>
`;
